### PR TITLE
feat: allow Doer interface for http.Service

### LINKED ISSUE #255: Should allow passing in an interface to http.Service
__Proposal:__
Instead of requiring http.Service to have a concrete http.Client, I would like it to take a `Doer` interface, since it only actually uses the `Do` method.

__Current behavior:__
Currently it takes a concrete `http.Client`, which is more difficult to implement against.

__Desired behavior:__
I have filed a small PR to implement this, that's probably easier than describing in more detail.

__Alternatives considered:__
Copying the `http.Service` implementation and creating my own.

__Use case:__
I want to use this module in the `flux` cli, which only passes around and interface wrapping aspects of an `http.Client`, not a full client.

### PR DESCRIPTION
**Pull request by lesam**
Closes #255

## Proposed Changes

See issue #255

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [ ] A test has been added if appropriate
- [x] Tests pass
- [x] Commit messages are in [semantic format](https://seesparkbox.com/foundry/semantic_commit_messages)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by lesam** [comment]
I need this to support this PR: https://github.com/influxdata/flux/pull/3718

**Review by vlastahajek — CHANGES_REQUESTED** [review]
Thanks for the PR!
In general, supporting Doer interface is a good enhancement 👍 

Implementation can be less complex by adding `SetHTTPDoer` and `HTTPDoer()` do api/http/Options.

**Comment by lesam** [comment]
I'm not sure that would be less complex.

We do need other methods (e.g. CloseIdleConnections) elsewhere, so we would have to error or panic when calling options.HTTPClient() if the client was not set up correctly - this seems worse than adding a constructor on Service.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## Unreleased
+
+### Features
+[#256](https://github.com/influxdata/influxdb-client-go/pull/256) struct http.Service can be created with a 'Doer' interface
```
Comment (vlastahajek) [inline CHANGELOG.md:4]:
```suggestion
[#256](https://github.com/influxdata/influxdb-client-go/pull/256) Allowing 'Doer' interface  for HTTP requests
```

File: api/http/options.go [hunk 2]
```diff
@@ -65,6 +71,32 @@ func (o *Options) OwnHTTPClient() bool {
...
+//
+// Setting the HTTPDoer will cause the other HTTP options
+// to be ignored.
+func (o *Options) SetHTTPDoer(d Doer) *Options {
+	if d == nil {
+		return o
+	}
+	o.doer = d
+	o.ownClient = false
+	return o
```
Comment (vlastahajek) [inline api/http/options.go:90]:
```suggestion
	if d != nil {
		o.doer = d
		o.ownClient = false
	}
	return o
```

File: api/http/options.go [hunk 3]
```diff
@@ -65,6 +71,32 @@ func (o *Options) OwnHTTPClient() bool {
...
+}
+
```
Comment (vlastahajek) [inline api/http/options.go:92]:
```suggestion
// HTTPDoer returns actual Doer if set, or http.Client
```

**Review by vlastahajek — CHANGES_REQUESTED** [review]
Thanks for the update. Just a couple of changes to-do yet.

**Review by vlastahajek — APPROVED** [review]

**Pull request merged**